usnistgov/oar-pdr-angular
# Support m1 builds, clean-up build scripts

This PR combines a number updates to the package's build scripts but which should not have any effect on the content of the actual built applications (but which require corresponding changes to oar-docker).  One key goal is to support builds for Macs featuring the m1 processor.  Changes include:
*  replace separate docker build containers `wizard` and `editable` with a single `build-test` container
*  uprev the gosu version used in the docker-based build (needed for m1 builds)
*  change distribution names, `wizard` and `editable` to `midas-author-wizard` and `midas-author-lps`
*  replace separate `makedist.editable` and `makedist.wizard` scripts with single `makedist` script
*  streamline docker-based build scripts
*  address some dependency vulnerabilities reported by `npm audit`

To test:
  1.  check out this branch
  2.  run `docker/dockbuild.sh` and ensure that it completes successfully
  3.  run `scripts/makedist.docker` and ensure that it completes successfully
  4.  ensure that the `dist` directory contains distributions (zip files and `_dep.json` files) with base names, `midas-author-wizard` and `midas-author-lps`
